app/models/paper.rb

```diff
@@ -12,6 +12,7 @@ class Paper < ApplicationRecord
 
   accepts_nested_attributes_for :authors, reject_if: proc { |attributes| attributes['name'].blank? }
   accepts_nested_attributes_for :links
+  validates_associated :links
```

Comment:
I understand, I just meant that in my testing, the validation was done whether or not this line was here.